Src/Common/RequestResponseHeaders.cs

```diff
@@ -8,12 +8,12 @@ internal static class RequestResponseHeaders
         /// <summary>
         /// Source instrumentation header that is added by an application while making http requests and retrieved by the other application when processing incoming requests.
         /// </summary>
-        public const string SourceInstrumentationKeyHeader = "x-ms-request-source-ikey";
```

Comment:
I'd then use more generic names at least. The idea is that target and source just grab the value of this header and pass it on... So when in fture it will be `appid-v2:` we will not need to change the logic of every SDK, we can gradually switch everything to v2

Also let's talk how we will include `roleInstance` in this header later. We shouldn't change a header name for it